### DISCUSSION TITLE
How do you use this with Meteor & config options?

### DISCUSSION THREAD

**Discussion by ajaypillay** [mdx-js/mdx#1570] [Q&A] [answered]
Hi, I am making use of this Meteor package, [apollinaire:mdx](https://github.com/Apollinaire/meteor-mdx) which gives the ability to import MDX files using this library.

Unfortunately this package doesn't support the passing of options, so I'm trying to hardcode it in, as I require LaTeX support through the `remark-math` and `rehype-katex` packages.

[I tried passing in the necessary config options in this line](https://github.com/Apollinaire/meteor-mdx/blob/master/compiler.js#L70) but when I console log `jsx`, I don't see the Math being parsed in my MDX file.

In [package.json](https://github.com/Apollinaire/meteor-mdx/blob/master/package.js), I have passed in the necessary dependencies so that it's aware of said packages.

I tried to `import remarkMath from "remark-math"` and when I console logged it in the constructor, it was defined but in the compile function it was undefined. I fixed this by changing it to `const remarkMath = require("remark-math")` and then it was defined in the compile function. Not really sure why this was the case.

But when I pass the config options in as such:
```
const jsx = mdx.sync(content, { remarkPlugins: [remarkMath], rehypePlugins: [rehypeKatex]})
```
And when I log `jsx`, I see everything being parsed correctly except the LaTeX example, which is copied verbatim from the [Math blocks page in the MDX docs](https://mdxjs.com/guides/math-blocks).

Just for sanity's sake - it's not the CSS that's being messed up, the Math is just contained within `<p>` tags and that's it, there's no added HTML tags.

How should I be setting this up correctly?

For more information:
1) I had to add the `babel-runtime` Meteor package and `@babel/runtime` NPM package, as it was complaining about those when I tried to add this custom package in my Meteor `packages` folder. I named this package `foo`.

**Comment by ChristianMurphy** [comment, marked as answer]
What version of MDX and remark math are you using?
MDX 1 needs remark-math <4.
MDX 2 needs remark-math >=4.

**Reply by ajaypillay** [reply]
Thank you for pointing this out. I'm still trying to debug this further. So at the time of opening this thread my versions were:
```js
 Package.registerBuildPlugin({
   name: 'compile-mdx',
   use: [
     'ecmascript',
     'babel-runtime',
     'babel-compiler@7.6.1',
     'caching-compiler@1.2.1',
   ],
   sources: [
     'compiler.js',
     'plugin.js',
   ],
   npmDependencies: {
     '@mdx-js/mdx': '1.6.22',
     "rehype-katex": "5.0.0",
     "remark-gfm": "1.0.0",
     "remark-math": "4.0.0",
     '@babel/runtime': '7.14.0'
   },
 });
```
But now I've changed the versions to:
```js
"rehype-katex": "4.0.0"
"remark-math": "3.0.1"
```
Now I can see the tags being output in my console but in my browser I'm getting the following error:
![image](https://user-images.githubusercontent.com/917752/120366515-54a09500-c342-11eb-926d-ea1351a35995.png)
![image](https://user-images.githubusercontent.com/917752/120366678-87e32400-c342-11eb-80f1-4e1e118373b8.png)

I'm still trying to figure this bit out and will update once I do.

I don't see any more `MDXTag` elements so I don't think I need that import anymore, looks like I need to import ~`MDXContent` or something else~ `mdx`?

I tried to add on the line `import mdx from "@mdx-js/mdx"` into `jsxComplete`:
```
const jsx = mdx.sync(content, { remarkPlugins: [remarkMath], rehypePlugins: [rehypeKatex]});
console.log(jsx);
const jsxComplete = `
import React from 'react'
import { MDXTag } from '@mdx-js/tag'
${jsx}
`;
```
But then my browser throws the error `Uncaught TypeError: [BABEL] unknown: Cannot read property 'explode' of undefined` and it gets thrown recursively so I have to close the tab. Not sure if this is a version issue as well, but I don't think it is.

**Reply by ajaypillay** [reply]
So this is the closest I've gotten so far.
```
import MDX from "./directory/test.mdx"
console.log(MDX)
```
This is what is logged in my browser:
![image](https://user-images.githubusercontent.com/917752/120376516-90d9f280-c34e-11eb-93bb-e2a7a3da20c2.png)

If I actually render the `<MDX />` component somewhere, I get the same error thrown on my server that `mdx` is not defined as per my server: `Error running template: ReferenceError: mdx is not defined` (I'm using SSR).

I see `mdx` in the return statement but it's not included anywhere in the file. Where do I need to include `mdx` from? I tried `import mdx from "@mdx-js/mdx"` and `const mdx = require("@mdx-js/mdx)` but they both cause the infinite error loop as described above.

**Reply by ChristianMurphy** [reply]
The loader includes the appropriate version of MDX and the framework for rendering (react, vue, preact, etc).
https://github.com/mdx-js/mdx/blob/907bd26d83287180c7155c0fb65671b7cb55c98d/packages/loader/index.js#L4-L37
It looks like meteor's is using outdated imports for MDX 1 https://github.com/Apollinaire/meteor-mdx/blob/0cee4c6ecc664c8a0341fee0cb1dff324b4fa14f/compiler.js#L73-L77

**Reply by ajaypillay** [reply]
Thank you! I removed the `MDXTag` import since it's not needed anymore and I included `import { mdx } from '@mdx-js/react'` and it works perfectly now.
